pkg/devfile/adapters/kubernetes/adapter.go

```diff
@@ -41,6 +41,39 @@ func (k Adapter) Push(parameters common.PushParameters) error {
 	return nil
 }
 
+// Build creates Kubernetes resources to build an image for the component
+func (k Adapter) Build(parameters common.BuildParameters) error {
+
+	err := k.componentAdapter.Build(parameters)
+	if err != nil {
+		return errors.Wrap(err, "Failed to build image for the component")
+	}
+
+	return nil
+}
+
+// Build creates Kubernetes resources to build an image for the component
+func (k Adapter) Deploy(parameters common.DeployParameters) error {
+
+	err := k.componentAdapter.Deploy(parameters)
+	if err != nil {
+		return errors.Wrap(err, "Failed to deploy the application")
+	}
+
+	return nil
+}
+
+// Build creates Kubernetes resources to build an image for the component
+func (k Adapter) DeployDelete(manifest []byte) error {
```

Comment:
instead of passing the whole manifest could we consider the minimum information that is necessary?

Reply:
Thanks for the feedback @girishramnani 
This may be better suited for #3478 as it was introduced there and just carried over into this PR